p34/tests/integration.rs

```diff
@@ -0,0 +1,177 @@
+pub mod integration {
+    use std::ops::{Add, Mul};
+
+    use p34::biguint::BigUint4096;
+    use p34::build_biguint_max_64;
+
+    #[test]
+    fn are_comparable() {
+        let expected = BigUint4096::new();
+
+        let actual = BigUint4096::new();
+
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn default_to_zero() {
+        let expected = BigUint4096::new();
+
+        let actual = BigUint4096::from([0; 64]);
+
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn add_one() {
+        let one = build_biguint_max_64!(1);
+
+        let actual = BigUint4096::new().add(one);
+
+        assert_eq!(actual, build_biguint_max_64!(1));
+    }
+
+    #[test]
+    fn add_one_to_all_limbs() {
+        let operand = BigUint4096::from([1; 64]);
+        let expected = BigUint4096::from([1; 64]);
+
+        let actual = BigUint4096::new().add(operand);
+
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn add_one_to_all_limbs_with_carry() {
+        let m = u64::MAX;
+        let mut _operand_1 = build_biguint_max_64!(m, m, 0, 0, 0, m, 0);
+        let _____operand_2 = build_biguint_max_64!(1, 0, 0, 0, 0, 1, 1);
+        let ______expected = build_biguint_max_64!(0, 0, 1, 0, 0, 0, 2);
+
+        let actual = _operand_1.add(_____operand_2);
+
+        assert_eq!(actual, ______expected);
+    }
+
+    #[test]
+    fn add_using_the_operator() {
+        let m = u64::MAX;
+        let mut _operand_1 = build_biguint_max_64!(m, m, 0, 0, 0, m, 0);
+        let _____operand_2 = build_biguint_max_64!(1, 0, 0, 0, 0, 1, 1);
+        let ______expected = build_biguint_max_64!(0, 0, 1, 0, 0, 0, 2);
```

Comment:
Such naming is not idiomatic. Underscore is usually used to mark unused variables.

Reply:
done